### DISCUSSION TITLE
Mozilla MDN Curriculum pre-launch announcement and RFC

### DISCUSSION THREAD

**Discussion by chrisdavidmills** [mdn/mdn-community#633] [Announcements]
The MDN front-end developer curriculum — talked about previously in multiple forums (for example, on the [MDN blog](https://developer.mozilla.org/en-US/blog/announcing-mdn-front-end-developer-curriculum/)) — aims to provide an up-to-date, industry-standard guide to the key skills front-end developers need to succeed in the web industry. It covers core technical skills, best practices, and modern tooling, and provides recommended learning resources. Note that the curriculum doesn’t include separate content to teach those skills — we felt that creating more competing content would be pointless, with all of the high-quality content already available.

In the future, Mozilla intends to include partner course recommendations and promote the curriculum as the basis for other educational resources such as further courses, certifications, etc.

The curriculum will be free for everyone to use.

## Release details

- The curriculum will be placed at https://github.com/mdn/curriculum/ and linked at the top level of the navigation. It is a separate resource from regular MDN content, in the same way as the Blog.
- The source will be contained in a separate repository from regular MDN content (you can already see a WIP version, at https://github.com/mdn/curriculum/tree/main/curriculum).  Mozilla welcomes feedback from anyone, in the form of issues filed on the repo.
Mozilla intends to release it in early- to mid-February.

## Why can’t it just be MDN content?

There are a few reasons for this:

- Mozilla envisaged the curriculum as a separate product from MDN content, which links to partner courses and a variety of learning resources as well as MDN. It therefore wouldn’t make sense to put it inside actual MDN content.
- It involves some specific custom design features that regular MDN content doesn’t provide.
- It contains Mozilla and Mozilla partner content that needs to be carefully controlled under the terms of partner agreements, etc. This would not fit comfortably with MDN content’s community-driven, consensus-seeking model.
- It is intended to be a slow-moving, stable curriculum, on which many other resources will base their structure. It therefore can’t be subject to the same level of rapid change as the main content.

**UPDATE**: A question was asked about whether curriculum-type content PRs on the main MDN content repo would be rejected because of competition with the curriculum. The Mozilla MDN curriculum team doesn't see this as a huge issue, given that traditionally MDN has not been a place to put curriculum-type content or learning pathways, and we are unlikely to start making a massive shift in this direction any time soon. They would like to discourage repetition of the type of curriculum treatment contained in the MDN Curriculum on regular MDN content — it would just compete needlessly, and not make sense. However, a structured set of guides or tutorials for teaching a specific technology or set of techniques would work fine in MDN content, and not complete with the curriculum. We could then potentially look at adding an extension module to the MDN Curriculum that would link to said MDN content as the place to go to learn that module. 

## Proposed changes to MDN content

At some point soon after the curriculum has been published, we intend to make some changes to MDN content, to allow the core content and the curriculum to complement each other more effectively and result in minimal confusion.

Key goals of this:

- Reduce confusion for readers over what the purpose of each resource is.
- Drive more traffic between the curriculum and the main MDN content.

### Changes we intend to make

1. Redirect the https://developer.mozilla.org/en-US/docs/Learn/Front-end_web_developer pathway to the MDN Curriculum. This was an attempt to provide a structured pathway for learning web development. But it wasn’t well thought out, hastily put together, and hasn’t been very effective.
  - Last year (Jan 2023-Jan 2024), the Learning Area had 4.40% of the total unique page views on the MDN References and Guides section. The Front-end web developer pathway had 1.39% of the Learning Area views (329,377 unique page views for the entire year).
  - In the last 30 days, Learn has 4.82% of the unique page views on MDN Docs References and Guides, with the pathway having 1.51% of this (25,170 views).
  - This is not nothing, but it is a small number of views considering how useful the page should be to people, and how many high-profile places it is linked from.
  - The page has a very short average time on page (1m 40s) and a fairly low exit % rate (29%).
  - This says to me that there is a need for such an organization of the learning area content. People are using the pathway to find other pages of interest, but they are not studying the page for very long and it is not very effective. This is also signified by the fact that 7 of the top 10 most viewed learning area pages are individual articles rather than learning area landing pages, which means the landing pages are probably not that effective in general and people are more likely to land on them using search.
  - I think this experience would be made much more successful by updating this function to use the much better thought-out MDN curriculum instead, driving a lot more traffic to the learning area in the process.
2. Update the learning area landing page content to make the purpose of the page clear, versus the MDN curriculum, and explain the interplay between the two.
  - The Learning area has never really been a structured curriculum/pathway. It is a useful collection of beginner’s articles.
  - Language like “This set of articles aims to guide complete beginners to web development with all that they need to start coding websites.” is not necessarily true and should better reflect how people might use the site.
3. Remove all the “Looking to become a front-end web developer?” boxes currently on MDN, and add back a much smaller number of links to the MDN Curriculum. For example, it would make sense to put one on the learning area landing page.
  - **UPDATE**: See https://github.com/mdn/content/issues/27244 for useful feedback about the confusion these boxes have caused.
  - **UPDATE**: We should think carefully about the placement of links to the MDN Curriculum from MDN content to make sure they do not cause confusion and are not perceived as spammy.
4. Update the https://developer.mozilla.org/en-US/docs/Learn#tasks_and_assessments section on the front page to make it less confusing, and somewhat deemphasize this aspect of the learning area.
5. Trim down or remove the https://developer.mozilla.org/en-US/docs/Learn#see_also links. For example:
  - Remove the https://www.youtube.com/playlist?list=PLo3w8EB99pqLEopnunz-dOOBJ8t-Wgt2g link, as it is 6 years old, and not going to be updated any time soon.
  - https://www.bitdegree.org/learn/ seems rather out of date, and very ad-heavy
  - Remove the https://foundation.mozilla.org/en/initiatives/web-literacy/core-curriculum/ link. The link doesn’t seem to go where you’d expect, and when you get there, many of the linked resources 404. Out of date.
6. Remove the MathML and Games content from the learning area and put it somewhere else. I know that they have been dumped here because we couldn’t figure out where else to put them, but they aren’t beginner’s content.
  - **UPDATE**: It has been requested that we wait on moving this content until we have had a chance to discuss it more carefully, probably as part of a bigger MDN IA/content structure conversation. This sounds fine and perfectly sensible; we are happy to defer this task.

## CTA

Please let us know what you think of this plan. We are happy to hear feedback.

**Comment by aardrian** [comment]
My only ask is that all partner content / curricula is / are accessible — as in, there is a commitment from each partner that it meets WCAG (at a minimum), includes an accessibility statement (with contact info), and provides an Accessibility Conformance Report (output of a VPAT).

I am making the assumption here that all MDN curriculum content already is.

**Reply by chrisdavidmills** [reply]
Hi @aardrian, lovely to hear from you!

A11y is important to me and important to Mozilla. The curriculum content will be accessible. As for partner content, a11y is top of mind for us as we search for partners and review their content.

It has been challenging to find vendors whose content

- is high-quality and engaging
- teaches web standards and best practices correctly and with a philosophy/approach that aligns with ours
- covers all the curriculum's core content and therefore conforms to the curriculum
- is accessible to an acceptable degree

Luckily we have found partners who are open and willing to work with us to improve their practices in these regards.

That said, what resources would you recommend passing them to read up more about accessibility statements and Accessibility Conformance Reports? The WAI stuff seems pretty good, and I usually send that to folks, but do you have any other resources to recommend?

**Reply by aardrian** [reply]
@chrisdavidmills I suspect part of the challenge of finding accessible third-party content is because few organizations demand it. Great opportunity for MDN to establish it as the price of admission.

> That said, what resources would you recommend passing them to read up more about accessibility statements and Accessibility Conformance Reports?

I imagine you know about the [WAI accessibility statement generator](https://www.w3.org/WAI/planning/statements/generator/#create) since you referenced WAI. It's what I used for my site and many client sites.

As for ACRs, the [ITIC site gives decent guidance](https://www.itic.org/policy/accessibility/vpat), but I have been making them for so long that I don't have other references handy. I do, however, feel strongly that an ACR _not_ generated by a third party should be treated warily. Organizations typically do not have the in-house expertise and are also motivated to self-report a rosier result.

More importantly, however, is knowing how to read an ACR and glean anything useful about actual accessibility versus reported accessibility. If you are interested in my thoughts on that, I have something around here I can paste into this convo.

**Comment by SphinxKnight** [comment]
Will the curriculum, intended for learners, be localizable?

If not, localizers for mdn/translated-content might need to reword the new “Looking to become a front-end web developer?” boxes with an adequate wording.

**Reply by chrisdavidmills** [reply]
Hi @SphinxKnight, lovely to hear from you too!

Initially, the curriculum will be launched in English only, but in the future we will provide capabilities to localize it, with the same processes as for regular MDN content (just for the locales we currently support).

This being said, we cannot control what localization our partners will have in place.

**Reply by Graywolf9** [reply]
Hi! Thanks @SphinxKnight for asking I also have interests on it, I'll stay tunned

**Comment by Jo-with-vision** [comment]
Hi,

This sounds like fantastic changes and very timely for me as I have just begun following the FE developer learning pathway in its current form. 

Adding the curriculum in the top-level navigation is a great decision - this will definitely help users (like me) to understand the difference between general reference documents and modular quick-guides on MDN and a full, structured FE curriculum. The biggest issue I've faced while following the current FE dev pathway (as mentioned in the issue linked in your update) is getting lost in the regular MDN content because the current sidebar navigation doesn't show me I am on the FE dev pathway. If the sidebar navigation for the curriculum is 'locked-in' to only the content within the new curriculum section, with clear highlights for where I currently am in within that context, this would be a great change for usability. 

RE: Guides section in general

When I first visited MDN, it was via the homepage. I clicked on 'Guides' immediately because I was seeking learning material, and that title made the most sense. I didn't spend long there however, because the callout advertising the FE pathway was high up on the first page, so I didn't actually read any of the other content on the first guide.

But having looked back at Guides now - I think the first page you're taken to upon clicking Guides 'Learn web development' could possibly be redundant with the addition of the curriculum section? Welcoming the user to the learning area is confusing if you have a new dedicated curriculum space. Guides is a space for your guides, right? I find it confusing when learning area is used to describe the section.

My feedback/suggestion:

Guides from the top-level navigation should lead you to a main content section with the corresponding header 'Guides'. The opening paragraph should welcome you to the guides section, explain the purpose of the section (eg it contains a series of short, helpful guides on topics from 'Getting started on the web' to introducing HTML, CSS and JavaScript and so on...). 

There could be a line underneath this directing users looking for more in depth-learning to the full curriculum section (or a callout if you guys just love callouts).

The rest of the contents of the page should just be the overview of the guides available on the sidebar. I wouldn't think any other content is necessary on this page once you have the curriculum section up and running. 

This would really streamline the Guides section and help users understand the difference between your collection of independent reference guides on the man MDN web docs and the curriculum section.  _note: I would also remove the heading 'complete beginners start here' from the guides sidebar navigation and replace it with something more meaningful like 'Guides in this section' or omit it entirely, again, just to make sure the guides section doesn't come across as the leaning area anymore._

Thanks so much for beginning the new curriculum  - I'm excited to see how it develops

**Reply by chrisdavidmills** [reply]
Thanks for all the feedback and extra context @Jo-with-vision , this is really helpful.

You are right that we do need to think about the wording on the learning area landing page and possibly even the name of the section, to make sure that its purpose is differentiated from that of the curriculum. I did cover that a bit in the above discussion text, but we will definitely think about it carefully as we make updates following the curriculum lunch.

Since the learning area is a set of guides for less experienced web developers, we could maybe even consider retitling it to something like "Beginner's guides", to help its function stand apart from both the curriculum and the rest of MDN. But that kind of change will need deeper discussion at a later date.

Thanks again for the comments. I'm looking forward to hearing what you think about the curriculum once it launches!

**Reply by Jo-with-vision** [reply]
Hi, thanks for your reply.

Yeah, you could name them beginner guides, but it would limit the entire section of guides to beginner content and you may consider changing that in the future. I think that's why I left my suggestion as just 'Guides'.

Some of your more difficult guides wouldn't seem out of place If the Guides section was just the place for all your guides, no matter the level. Will the launch of a linear curriculum aimed at beginners mean you still need your guides section to only contain beginner content, or could it contain a broader range of guides?

I will be interested to hear the results of your deeper discussions about it all later on. 

I am sure it's a typo, but a curriculum lunch sounds like it would be a great perk to your project!

**Comment by chrisdavidmills** [comment]
A small update, just for the record.

I discussed this plan with @Elchi3, and he shared some feedback that him and the rest o the Open Web Docs folks had about it. Florian, please chime in if you have anything else to add to this description.

The three points they had were, to summarize, as follows:

1. They raised a concern about whether curriculum-type content would be rejected from the MDN content repo in the future, because of worries about it competing with the MDN Curriculum content. I added the following note above to address this:

  > UPDATE: A question was asked about whether curriculum-type content PRs on the main MDN content repo would be rejected because of competition with the curriculum. The Mozilla MDN curriculum team doesn't see this as a huge issue, given that traditionally MDN has not been a place to put curriculum-type content or learning pathways, and we are unlikely to start making a massive shift in this direction any time soon. They would like to discourage repetition of the type of curriculum treatment contained in the MDN Curriculum on regular MDN content — it would just compete needlessly, and not make sense. However, a structured set of guides or tutorials for teaching a specific technology or set of techniques would work fine in MDN content, and not complete with the curriculum. We could then potentially look at adding an extension module to the MDN Curriculum that would link to said MDN content as the place to go to learn that module.

2. They raised a concern about how the MDN Curriculum would be linked to, given that it will contain promotional links, and the Mozilla team may wish to link to it from a lot of places on MDN content, which may come across as spammy/promotional. I've updated the wording in the third main bullet down (the one that starts with "Remove all the “Looking to become a front-end web developer?” boxes..." to address this.

3. They asked whether we could wait to make structural changes to MDN content until we've had a chance to have a wider discussion about it. To address this, I added the following note above:

  > UPDATE: It has been requested that we wait on moving this content until we have had a chance to discuss it more carefully, probably as part of a bigger MDN IA/content structure conversation. This sounds fine and perfectly sensible; we are happy to defer this task.

**Comment by wbamberg** [comment]
I understand that this bit is on hold, but:

> Remove the MathML and Games content from the learning area and put it somewhere else. I know that they have been dumped here because we couldn’t figure out where else to put them, but they aren’t beginner’s content. 

This is true for Games, for sure, but MathML was certainly [written from the beginning as a part of the Learn area](https://github.com/mdn/content/pull/19467) and is absolutely properly integrated into it. We might want to say it doesn't belong because Learn is for beginners, but the reason it was added there is that there is nowhere else on MDN that is "the obvious place" to put structured learning material, whether it's beginner or advanced.

**Reply by chrisdavidmills** [reply]
What you say here is reasonable. I guess it just irks me when I see it there because I feel like the original intention of `/Learn` was that it would be for beginners to web development, and I always wrote content for it from that perspective. From that point of view, it feels really out of place.

But then again, the intention mentioned above was not really discussed that much with others, and quite a bit of other stuff inside `/Learn` is also quite advanced, so not exactly for beginners ;-)

Maybe a new focus for `/Learn` would be "Beginner guides for web development technologies" (but not necessarily for complete beginners to web development). Definitely something we could discuss at the document structure/IA workshop.

**Comment by pransh15** [comment]
Closing this since it's been shipped, and resolved.
